### ISSUE TITLE
value-keyword-case: `currentcolor`should be acceptable

### ISSUE THREAD

**Issue by silverwind** [stylelint/stylelint#4884]
> Clearly describe the bug

`currentcolor` (all lowercase) should be accepted. CSS Color Module level 4 defines it like that.

https://www.w3.org/TR/css-color-4/#currentcolor-color

Some historical context from above link:

> it was originally introduced in SVG as a special attribute value spelled "currentColor", rather than a CSS value. Only later did CSS pick it up, at which point the capitalization stopped mattering, as CSS keywords are case-insensitive.

> Which rule, if any, is the bug related to?

`value-keyword-case`

> What code is needed to reproduce the bug?

```css
a {
  color: currentcolor;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "value-keyword-case": "lower"
  }
}
```

> Which version of stylelint are you using?

e.g. `13.6.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

No error.

> What actually happened (e.g. what warnings or errors did you get)?

```shell
test.css
 23:43  ✖  Expected "currentcolor" to be "currentColor"   value-keyword-case
```

**Comment by malsf21** [comment]
Hi @silverwind,

Thank you for reporting this edge case, and for using the issue template. I can confirm that this issue exists with [this demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYAdAdilkA2EATvLgK6GEyYAu+RA3FgL5Zaoba4QHEp4CWAcwAW1AM4wkgylUaYmIADQgxATxpIAHgnBixS8BEwAzITo450IQqTwwxVkhZworANyR5SMALQBrGFUAdyIoHzAkCUdXEAIgmEIrLBwWeRAmIA), and contrast it to another keyword that is all-lowercase even though it is multiple english words (in this case, `lightseagreen`).

Solving this problem should be pretty easy: I think we'd just need to check for `currentcolor` in this sea of ignore checks

https://github.com/stylelint/stylelint/blob/30bacfc2b6559b8428d44c01e612e343021e6be7/lib/rules/value-keyword-case/index.js#L59-L85

(note how `lightseagreen` is ignored because it's in the `keywordSet` called `systemColors`)

In terms of actual implementation, there are several approaches we could take here. I think a simple one would be to just add a manual check to `currentcolor`, though there may be more efficient solutions. I'll let other members of the team chime in with more information.

In the meantime, I encourage you to take a crack at implementing this fix if you can! You can find out more information on getting started and fixing a bug in the rule [on the Stylelint docs page](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule). I'll label this issue accordingly, and let us know if you have any questions!

**Comment by jeddy3** [comment]
@silverwind Thanks for the report and for linking to the spec.

> In terms of actual implementation, there are several approaches we could take here. I think a simple one would be to just add a manual check to currentcolor,

The rule is suggesting `currentColor` because it's included within the [set of camel case keywords](https://github.com/stylelint/stylelint/blob/master/lib/reference/keywordSets.js#L327) that the rule uses.

As the specification is now advocating the use of a lowercase `currentcolor` I think we can make this the default behaviour _for all of these SVG-based keywords_. I think we'll need to offer an option for users wanting to camel case them, though.

```
camelCaseSvgKeywords: true | false (default false)
```

I've labelled the issue as ready to implement. @silverwind Please consider [contributing](https://stylelint.io/contributing) if you have time.  It should just be the case of moving the existing camelCase behaviour behind the new option.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.

**Comment by silverwind** [comment]
I'm still wondering why a the value of `lower` for `value-keyword-case` enforces camelCase. It seems the rule should either enforce what the user specifies or gain a `camelcase` option.

**Comment by jeddy3** [comment]
> I'm still wondering why the value of `lower` for `value-keyword-case` enforces camel case.

We thought that a [small set of keywords that come from SVG-land](https://github.com/stylelint/stylelint/blob/master/lib/reference/keywordSets.js#L327), e.g. `text-rendering` values like [`optimizeLegibility`](https://developer.mozilla.org/en-US/docs/Web/CSS/text-rendering#Syntax), were always meant to be camel case, regardless of whether all the other keywords were lower (or upper) case. The change in the spec illustrates this isn't the case anymore.

> I think the rule should either enforce what the user specifies or gain a camelcase option.

Yes, that's [the proposal](https://github.com/stylelint/stylelint/issues/4884#issuecomment-671072108).  We change the default behaviour so that the primary option (`lower` or `upper`) is used for every keyword, and add a `camelCaseSvgKeywords: true` option for users who want the existing behaviour.

**Comment by ZoeBijl** [comment]
Question: the [CSS Color Level 3](https://www.w3.org/TR/css-color-3/#currentcolor) (current recommendation) still lists this as `currentColor`. Should stylelint follow that instead of documents that are still in draft?
